src/main/java/subway/repository/DbLineRepository.java

```diff
@@ -0,0 +1,157 @@
+package subway.repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Repository;
+import subway.dao.LineDao;
+import subway.dao.StationEdgeDao;
+import subway.domain.Line;
+import subway.domain.StationEdge;
+import subway.entity.LineEntity;
+import subway.entity.StationEdgeEntity;
+import subway.exception.StationEdgeNotFoundException;
+
+
+@Repository
+public class DbLineRepository implements LineRepository {
+
+    private final LineDao lineDao;
+    private final StationEdgeDao stationEdgeDao;
+
+
+    public DbLineRepository(LineDao lineDao, StationEdgeDao stationEdgeDao) {
+        this.lineDao = lineDao;
+        this.stationEdgeDao = stationEdgeDao;
+    }
+
+    @Override
+    public Optional<Line> findById(Long id) {
+        Optional<LineEntity> optionalLineEntity = lineDao.findById(id);
+        if (optionalLineEntity.isEmpty()) {
+            return Optional.empty();
+        }
+
+        LineEntity lineEntity = optionalLineEntity.get();
+
+        return convertToDomain(lineEntity);
+    }
+
+    private Optional<Line> convertToDomain(LineEntity lineEntity) {
+        List<StationEdgeEntity> stationEdgeEntities = stationEdgeDao.findByLineId(lineEntity.getId());
+        List<StationEdge> sorted = getStationEdges(stationEdgeEntities);
+
+        return Optional.of(Line.of(lineEntity.getId(), lineEntity.getName(), lineEntity.getColor(), sorted));
+    }
+
+    private List<StationEdge> getStationEdges(List<StationEdgeEntity> stationEdgeEntities) {
+        int numberOfStations = stationEdgeEntities.size();
+        List<StationEdge> sorted = new ArrayList<>();
+        StationEdgeEntity first = getFirstStationEdgeEntity(stationEdgeEntities);
+        sorted.add(first.toDomain());
+        Long previousId = first.getId();
+
+        while (sorted.size() < numberOfStations) {
+            StationEdgeEntity next = getNext(stationEdgeEntities, previousId);
+            sorted.add(next.toDomain());
+            previousId = next.getId();
+        }
+        return sorted;
+    }
+
+    private static StationEdgeEntity getFirstStationEdgeEntity(List<StationEdgeEntity> stationEdgeEntities) {
+        StationEdgeEntity first = stationEdgeEntities.stream()
+                .filter(stationEdgeEntity -> stationEdgeEntity.getPreviousStationEdgeId() == null)
+                .findFirst()
+                .get();
+        stationEdgeEntities.remove(first);
+        return first;
+    }
+
+    private static StationEdgeEntity getNext(List<StationEdgeEntity> stationEdgeEntities,
+                                             Long previousId) {
+        StationEdgeEntity next = stationEdgeEntities.stream()
+                .filter(stationEdge -> stationEdge.getPreviousStationEdgeId().equals(previousId))
+                .findFirst().get();
+        return next;
```

Comment:
별건 아니지만, ealry retrun을 해보면 어떨까 해요.
여기도 get을 그냥 쓰네요 ㅠㅠ
```suggestion
        return stationEdgeEntities.stream()
                .filter(stationEdge -> stationEdge.getPreviousStationEdgeId().equals(previousId))
                .findFirst().get();```